autofaiss/indices/distributed.py

```diff
@@ -336,3 +342,315 @@ def run(
+    max_index_memory_usage: str = "16G",
+    min_nearest_neighbors_to_retrieve: int = 20,
+    embedding_column_name: str = "embedding",
+    current_memory_available: str = "32G",
+    use_gpu: bool = False,
+    metric_type: str = "ip",
+    nb_cores: Optional[int] = None,
+    make_direct_map: bool = False,
+) -> Tuple[Optional[faiss.Index], Optional[Dict[str, str]]]:
+    """
+    Create a small index
+    """
+
+    # Train index
+    trained_index = create_and_train_index_from_embedding_dir(
+        embedding_root_dir=embedding_root_dir,
+        embedding_column_name=embedding_column_name,
+        max_index_memory_usage=max_index_memory_usage,
+        make_direct_map=make_direct_map,
+        should_be_memory_mappable=should_be_memory_mappable,
+        use_gpu=use_gpu,
+        metric_type=metric_type,
+        nb_cores=nb_cores,
+        current_memory_available=current_memory_available,
+        id_columns=id_columns,
+    )
+
+    # Add embeddings to index and compute metrics
+    return _add_embeddings_to_index(
+        add_embeddings_fn=partial(
+            add_embeddings_to_index_local,
+            trained_index_or_path=trained_index.index_or_path,
+            add_embeddings_with_ids=True,
+        ),
+        embedding_reader=trained_index.embedding_reader_or_path,
+        output_root_dir=output_root_dir,
+        index_key=trained_index.index_key,
+        current_memory_available=current_memory_available,
+        id_columns=id_columns,
+        max_index_query_time_ms=max_index_query_time_ms,
+        min_nearest_neighbors_to_retrieve=min_nearest_neighbors_to_retrieve,
+        use_gpu=use_gpu,
+        make_direct_map=make_direct_map,
+    )
+
+
+def create_partitioned_indexes(
+    partitions: List[str],
+    big_index_threshold: int,
+    output_root_dir: str,
+    nb_cores: Optional[int],
+    nb_splits_per_big_index: int,
+    id_columns: Optional[List[str]] = None,
+    max_index_query_time_ms: float = 10.0,
+    min_nearest_neighbors_to_retrieve: int = 20,
+    embedding_column_name: str = "embedding",
+    max_index_memory_usage: str = "16G",
+    current_memory_available: str = "32G",
+    use_gpu: bool = False,
+    metric_type: str = "ip",
+    make_direct_map: bool = False,
+    should_be_memory_mappable: bool = False,
+    temp_root_dir: str = "hdfs://root/tmp/distributed_autofaiss_indices",
+) -> List[Optional[Dict[str, str]]]:
+    """
+    Create partitioned indexes from a list of parquet partitions,
+    i.e. create and train one index per parquet partition
+    """
+
+    def _create_small_indexes(embedding_root_dirs: List[str]) -> List[Optional[Dict[str, str]]]:
+        rdd = ss.sparkContext.parallelize(embedding_root_dirs, len(embedding_root_dirs))
+        return rdd.map(
+            lambda embedding_root_dir: create_small_index(
+                embedding_root_dir=embedding_root_dir,
+                output_root_dir=output_root_dir,
+                id_columns=id_columns,
```

Comment:
FAR: Maybe we should put all the parameters in dataclasses, because it is hard to verify if on parameter is missing